### ISSUE TITLE
`use_build_context_synchronously` false negative when checking mounted for different BuildContext

### ISSUE THREAD

**Issue by AhmedLSayed9** [dart-lang/linter#5123]
**Describe the issue**
When using multiple `BuildContext`, `use_build_context_synchronously` is not triggered if I checked for `.mounted` for a different BuildContext.

**To Reproduce**
```dart
Future<void> f1(BuildContext context, BuildContext anotherContext) async {
  await Future<void>.delayed(Duration.zero);
  // print(context.size);  //this triggers fine
  if (anotherContext.mounted) print(context.size);
}
```

**Expected behavior**
The linter should trigger as I've not checked for `context.mounted`.

**Comment by srawlins** [comment]
Good catch 😁  Don't expect this to be fixed any time soon. I think I'm just going to document this as a limitation of the rule.
